### ISSUE TITLE
Support async providers

### ISSUE THREAD

**Issue by ch1ller0** [ch1ller0/fridgefm-inverter#39]
* **I'm submitting a ...**
[ ] bug report
[x] feature request
[ ] question about the decisions made in the repository
[ ] question about how to use this project

* **Summary**
It is nice to have support of async providers, so that it is possible to wait for some async task to be completed before starting the application. https://docs.nestjs.com/fundamentals/async-providers

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. StackOverflow, personal fork, etc.)

**Comment by ch1ller0** [comment]
Closes via #40
